Extension/src/LanguageServer/configurations.ts

```diff
@@ -736,34 +737,57 @@ export class CppProperties {
     }
 
     private resolveAndSplit(paths: string[] | undefined, defaultValue: string[] | undefined, env: Environment, glob = false): string[] {
-        const result: string[] = [];
-        if (paths) {
-            paths = this.resolveDefaults(paths, defaultValue);
-            paths.forEach(entry => {
-                const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
-                result.push(...entries);
-            });
-        }
+        const resolvedVariables: string[] = [];
+        if (paths === undefined) {
+            return resolvedVariables;
+        }
+        paths = this.resolveDefaults(paths, defaultValue);
+        paths.forEach(entry => {
+            const entries: string[] = util.resolveVariables(entry, env).split(util.envDelimiter).map(e => this.resolvePath(e, false)).filter(e => e);
+            resolvedVariables.push(...entries);
+        });
         if (!glob) {
-            return result;
-        }
-
-        const globResult: string[] = [];
-        for (let res of result) {
-            // fastGlob will expand the ending double wildcard. temporary strip them before expanding
-            const recursive: boolean = res.endsWith('**');
-            if (recursive) {
-                res = res.slice(0, res.length - 2);
+            return resolvedVariables;
+        }
+        const resolvedGlob: string[] = [];
+        for (let res of resolvedVariables) {
+            let counter: number = 0;
+            let slashFound: boolean = false;
+            const lastIndex: number = res.length - 1;
+            // Detect and glob all wildcard variations by looking at last character in the path first.
+            for (let i: number = lastIndex; i >= 0; i--) {
+                if (res[i] === '*') {
+                    counter++;
+                } else if (res[i] === '/' || (res[i] === '\\' && isWindows)) {
+                    counter++;
+                    slashFound = true;
+                    break;
+                } else {
+                    break;
+                }
+            }
+            let suffix: string = '';
+            if (slashFound) {
+                suffix = res.slice(res.length - counter);
+                res = res.slice(0, res.length - counter);
             }
-            // fastGlob can't deal with backslash-separated path => remove them
-            const normalized: string = res.replace(/\\/g, '/');
-            const cwd: string = this.rootUri?.fsPath?.replace(/\\/g, '/') || '';
+            let normalized: string = '';
+            let cwd: string = '';
```

Comment:
Isn't this going to leave `cwd` a blank string?  Shouldn't it be set to `this.rootUri?.fsPath || ''`?

Similarly for `normalized`.  Shouldn't that be set to `res` when not on Windows?

Also, shouldn't all references to `res` below be referencing `normalized`?